### PR TITLE
django_manage - added splitting the command parameter for running

### LINKED ISSUE #3215: community.general.django_manage fails on commands with arguments
### Summary

Since #3098 it's impossible to run django commands with arguments, like
```sh
./manage.py collectstatic --noinput
```
The whole `collectstatic --noinput` string is considered one argument now, which results in an error.

### Issue Type

Bug Report

### Component Name

django_manage

### Ansible Version

4.4.0

### Steps to Reproduce

```yaml
- name: Run collectatatic
  community.general.django_manage:
    command: collectstatic --noinput
    project_path: "{{ django_dir }}"
```

### Expected Results

I expect the command to be executed as `["./manage.py", "collectstatic", "--noinput"]`

### Actual Results

```
fatal: [example.com]: FAILED! => changed=false 
  cmd:
  - ./manage.py
  - collectstatic --noinput
  msg: |-
    stdout: Unknown command: 'collectstatic --noinput'
    Type 'manage.py help' for usage.
  
    :stderr: Unknown django command: collectstatic --noinput
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Added splitting generic CLI arguments from the `command` parameter

Fixes #3215 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/modules/web_infrastructure/django_manage.py

### CODE REVIEW

**Comment by russoz** [comment]
@314eter it took a bit longer than expected, hope it does the trick.

File: plugins/modules/web_infrastructure/django_manage.py [hunk 1]
```diff
@@ -288,9 +290,9 @@ def main():
 
     _ensure_virtualenv(module)
 
-    cmd = ["./manage.py", command]
+    cmd = ["./manage.py"] + command_split
 
-    if command in noinput_commands:
+    if command in noinput_commands and '--noinput' not in command_split:
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/django_manage.py:295]: Should it be
```suggestion
    if command[0] in noinput_commands and '--noinput' not in command_split:
```
? (Or can options also be before it, and we need something like `any(cmd in noinput_commands for cmd in command)`?

Comment (russoz) [reply plugins/modules/web_infrastructure/django_manage.py:295]: Errr..... no. See lines 277-278. The expanded array with all arguments is in `command_split` whilst `command` already contains only the first element. ;-)

Comment (russoz) [reply plugins/modules/web_infrastructure/django_manage.py:295]: And writing that, I reckon I could refactor all these variables to clearer names ;-)

**Comment by russoz** [comment]
(Cc: @wadevries)

File: plugins/modules/web_infrastructure/django_manage.py (hunk 1 above)
Comment (felixfontein) [reply plugins/modules/web_infrastructure/django_manage.py]: Ah, right. I guess it should be `... command_split[0] in noinput_commands ...` then :) (Or one of the `command_split` parts?)

Comment (russoz) [reply plugins/modules/web_infrastructure/django_manage.py]: I changed the variable `command` to `command_bin`, and its value ***is*** `command_split[0]`, so from where I am standing, it looks like the code is already doing what you're saying.

Comment (felixfontein) [reply plugins/modules/web_infrastructure/django_manage.py]: Ah, I forgot about that last part when writing the previous comment! Everything's fine now :)

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for fixing this!

**Comment by pandafy** [comment]
Hi! Will this not be backported to stable-2 and stable-3 ?

**Comment by felixfontein** [comment]
It has been backported to stable-3. The automatic backport to stable-2 failed, so if nobody wants to backport it manually, it won't be backported to stable-2.

**Pull request merged**